src/app/static/src/app/store/basic/basic.ts

```diff
@@ -0,0 +1,22 @@
+import { StoreOptions } from "vuex";
+import { AppState } from "../AppState";
+
+export interface BasicState extends AppState {
+    basicProp: string
+}
+
+const defaultState: BasicState = {
+    title: "Basic App",
+    appType: "basic",
+    basicProp: "basic value"
+};
+
+// Injected into view by server
+declare const appConfig: Partial<BasicState>;
```

Comment:
This pattern will be a little fiddly to test; probably a more "Vuex" way to do it is to fetch data from the server via an api call on mount. Or for properties that don't actually need to end up in the store (like app title), to pass these as props to the root component in the handlebar template:

```
<div id="app">
  <basic-app :title="{{ config.title }}"></basic-app>
</div>
```